cpp/src/io/csv/csv_gpu.cu

```diff
@@ -649,12 +652,8 @@ CUDF_KERNEL void __launch_bounds__(rowofs_block_dim)
                          int escapechar,
                          int commentchar)
 {
-  auto start         = data.begin();
-  using block_reduce = typename cub::BlockReduce<uint32_t, rowofs_block_dim>;
-  __shared__ union {
-    typename block_reduce::TempStorage bk_storage;
-    __align__(8) uint64_t ctxtree[rowofs_block_dim * 2];
```

Comment:
Out of curiosity, would this fix the misaligned access issue?
```cpp
  __shared__ __align__(16) union {		
    typename block_reduce::TempStorage bk_storage;		
     uint64_t ctxtree[rowofs_block_dim * 2];
```

Reply:
I don't think I tried this exactly, but I tried separating `bk_storage` and `ctxtree`, while keeping the alignment as-is
Also tried with `__align__(16)`, but probably still within the union. I can run your version tomorrow.